src/sql/parts/profiler/contrib/profilerActions.contribution.ts

```diff
@@ -46,3 +50,48 @@ CommandsRegistry.registerCommand({
 		return editorService.openEditor(profilerInput, { pinned: true }, false).then(() => TPromise.as(true));
 	}
 });
+
+KeybindingsRegistry.registerCommandAndKeybindingRule({
+	id: 'profiler.newProfiler',
+	weight: KeybindingsRegistry.WEIGHT.builtinExtension(),
+	when: undefined,
+	primary: KeyMod.CtrlCmd | KeyMod.Alt | KeyCode.KEY_P,
+	mac: { primary: KeyMod.CtrlCmd | KeyMod.Alt | KeyCode.KEY_P },
+	handler: CommandsRegistry.getCommand('profiler.newProfiler').handler
+});
+
+CommandsRegistry.registerCommand({
+	id: 'profiler.start',
+	handler: (accessor: ServicesAccessor) => {
+		let profilerService: IProfilerService = accessor.get(IProfilerService);
+		let editorService: IWorkbenchEditorService = accessor.get(IWorkbenchEditorService);
+
+		let activeEditor = editorService.getActiveEditor();
+		if (activeEditor instanceof ProfilerEditor) {
+			let profilerInput = activeEditor.input;
+			return profilerService.startSession(profilerInput.id).then(() => {
+				profilerInput.state.change({ isRunning: true, isStopped: false, isPaused: false });
+				return true;
+			});
+		}
+		return TPromise.as(false);
+	}
+});
+
+CommandsRegistry.registerCommand({
+	id: 'profiler.stop',
+	handler: (accessor: ServicesAccessor) => {
+		let profilerService: IProfilerService = accessor.get(IProfilerService);
+		let editorService: IWorkbenchEditorService = accessor.get(IWorkbenchEditorService);
+
+		let activeEditor = editorService.getActiveEditor();
+		if (activeEditor instanceof ProfilerEditor) {
+			let profilerInput = activeEditor.input;
+			return profilerService.stopSession(profilerInput.id).then(() => {
+				profilerInput.state.change({ isStopped: true, isPaused: false, isRunning: false });
```

Comment:
Is there some way that the call to `profilerService.stopSession` could automatically update the `profilerInput` state? If we want to update the input state every time we call start or stop then it would be easier for the `profilerService` to handle it instead of needing to have this state change logic here